kotlin-plugin/src/main/kotlin/me/shika/generation/utils.kt

```diff
@@ -40,7 +40,7 @@ fun IrClass.needSerializableFix(): Boolean {
 
 fun IrClass.isSerializable(): Boolean {
     return allSuperInterfaces()
-        .any { it.fqNameWhenAvailable == SERIALIZABLE_FQ_NAME || it.isSerializable() }
+        .any { it.fqNameWhenAvailable == SERIALIZABLE_FQ_NAME || it !== this && it.isSerializable() }
```

Comment:
I'm not sure why, but scala traits have themselves as a superinterface. This leads to StackOverflow errors when the trait does not implement `Serializable`.